agent/handlers/v3/task_metadata_handler.go

```diff
@@ -53,11 +54,24 @@ func TaskMetadataHandler(state dockerstate.TaskEngineState, ecsClient api.ECSCli
 
 		taskResponse, err := v2.NewTaskResponse(taskARN, state, ecsClient, cluster, az, containerInstanceArn, propagateTags, false)
 		if err != nil {
-			errResponseJSON, err := json.Marshal("Unable to generate metadata for task: '" + taskARN + "'")
+			statusCode := http.StatusInternalServerError
+			errorResponse := fmt.Sprintf("Unable to generate metadata for v3 task: '%s'.", taskARN)
+
+			if awsErr, ok := err.(awserr.Error); ok {
+				if reqErr, ok := err.(awserr.RequestFailure); ok {
+					statusCode = reqErr.StatusCode()
+				}
+				errorLogMessage := fmt.Sprintf(" Received the error code: %s, the error message: %s,"+
```

Comment:
can you move this inside the 'if' condition since we may not want to necessarily throw internal server error if status code is empty?

Reply:
Moved `errorLogMessage` into the `if` statement, as shown [here](https://github.com/aws/amazon-ecs-agent/blob/991ff6eb2bab4386e43c762e5b331725c90ebc5d/agent/handlers/v4/task_metadata_handler.go#L57-L64). Thanks!